### PR TITLE
Wrong rendering of Markdown files

### LINKED ISSUE #9951: [BUG] Wrong rendering of Markdown files
### Rancher Server Setup

- Rancher version: 2.7.6

### Information about the Cluster

N/A

### User Information

N/A

### Describe the bug

Markdown files are rendered in a wrong way in the Apps section.

### To Reproduce

- Go to Apps, click on Charts
- Select an app

### Result

A `README` file is rendered but some elements (especially tables) are wrong. The information is unreadable.

### Expected Result

- The output could be improved in terms of margins, sizes (headers, paragraphs), etc...
- Tables are rendered correctly.

### Screenshots

screenshot1

### Additional context

A good render can improve a lot the readability and user experience.

screenshot2

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #9951 

### Occurred changes and/or fixed issues

- remove `DOMpurify` from markdown component (`v-clean-html` does exactly this, so need to duplicate it)
- add missing `html` tags to allowed tags in `v-clean-html` so that markdown content is rendered correctly

**NOTE:** links in markdown won't open in a new tab unless https://github.com/rancher/dashboard/pull/9928 is merged as well.

### Areas or cases that should be tested

- Go to `https://<!--- RANCHER BASE URL --->/c/local/apps/charts/chart?repo-type=cluster&repo=rancher-partner-charts&chart=tomcat&version=10.10.10` and check that markdown is rendered correctly

### Screenshot/Video

Screenshot 2023-10-20 at 11 07 18
Screenshot 2023-10-20 at 11 07 22

### CODE REVIEW

**Comment by recena** [comment]
@aalves08 Thank you. It seems tables do not have styles (look & feel).

File: shell/plugins/clean-html-directive.js [hunk 1]
```diff
@@ -15,6 +15,18 @@ const ALLOWED_TAGS = [
...
+  'h3',
+  'h4',
+  'h5',
+  'h6',
+  'table',
+  'thead',
+  'tr',
+  'th',
+  'tbody',
+  'td',
```
Comment (recena) [inline shell/plugins/clean-html-directive.js:29]: Why not `blockquote` for example?

**Review by richard-cox — COMMENTED** [review]
@rak-phillip can you take a look at this one? I think it's mac specific, i couldn't reproduce at all.

Changes generally look ok. The custom instance of dompurity used in markdown component is removed and it now uses the global one from the clean directive.

**Comment by aalves08** [comment]
@rak-phillip take into account that changes from https://github.com/rancher/dashboard/pull/9928 need to go in and this PR will need to be adjusted accordingly

File: shell/plugins/clean-html-directive.js (hunk 1 above)
Comment (aalves08) [reply shell/plugins/clean-html-directive.js:29]: will add it... This PR needs to be revisited as it's dependent of #9928 , which will lead to changes here.

**Comment by rak-phillip** [comment]
@aalves08 All in all, I think the change looks good.

However, I'm testing against 2.8-head and I can't reproduce the problem as described in the issue. For example, I see the output rendered as expected while reviewing the Apache Tomcat chart

image

Is there an environment where I can see this issue in action so that we can validate that the change fixes the problem?

**Comment by aalves08** [comment]
@rak-phillip I was able to repro this on a `2.7.6` system like the original issue. I'm running a `2.8.0-rc3` right now and I cannot repro it, which is strange. I don't know what changed 🤔

**Comment by aalves08** [comment]
@rak-phillip can I get an approval here? I've just updated the PR after the merge of https://github.com/rancher/dashboard/pull/9928 by Neil

**Review by rak-phillip — APPROVED** [review]
I think this change looks good. I cannot reproduce the original issue. Even still, I think this change makes sense.

**Pull request merged**